src/hooks/useRelayerData/useRelayerData.tsx

```diff
@@ -12,12 +12,14 @@ const RESET_DATA_AFTER = 250
 export default function useRelayerData({
   fetch,
   url,
-  initialState = null
+  initialState = null,
+  useOfflineStatus = null
 }: UseRelayerDataProps): UseRelayerDataReturnType {
   const [isLoading, setLoading] = useState<boolean>(true)
   const [data, setData] = useState<any>(initialState)
   const [err, setErr] = useState<any>(null)
   const prevUrl = useRef('')
+  const isOffline = (!!useOfflineStatus && useOfflineStatus()) || false
```

Comment:
Why pass the entire hook, call it and get the value here, when you can just pass isOffline and give it a default value of false? `isOffline=false`

You are importing it in the file that you call `useRelayerData` anyway, so just call it there, get the value of isOffline and pass it to the hook.